moto/iam/models.py

```diff
@@ -1093,19 +1097,29 @@ def __init__(
         self.status = status
         self.create_date = utcnow()
 
+        self._last_used: Optional[AccessKeyLastUsed] = None
+        self.role_arn: Optional[str] = None
+
+    @property
+    def last_used(self) -> Optional[AccessKeyLastUsed]:
+        return self._last_used
+
+    @last_used.setter
+    def last_used(self, value: Union[datetime, AccessKeyLastUsed, None]) -> None:
         # Some users will set this field manually
         # And they will be setting this value to a `datetime`
         # https://github.com/getmoto/moto/issues/5927#issuecomment-1738188283
-        #
-        # The `to_csv` method calls `last_used.strptime`, which currently works on both AccessKeyLastUsed and datetime
-        # In the next major release we should communicate that this only accepts AccessKeyLastUsed
-        # (And rework to_csv accordingly)
-        self.last_used: Optional[AccessKeyLastUsed] = None
```

Comment:
I'd like to keep this section as is, as well. This improvement does make sense, but it is indeed a breaking change, and I know there are other people who set this field directly - so I only want to change the behaviour in a major version.

My current plan to improve this: let users create an `AccessKeyLastUsed`-object and set that directly, which gives them explicit control over the date-field _and_ the account/region.
It's a few more lines of code, but it is more flexible. Would that work for you?

Reply:
Makes sense to me. Reverted.

Reply:
For what it's worth I'm already doing that in the relevant code: import both iam_backends and AccessKeyLastUsed, create the AccessKeyLastUsed object and set it. 

I thought it might be clever to allow users to skip that and set a datetime object but get a proper AccessKeyLastUsed anyway, but I can also understand why that might be a little too clever and just be confusing.